### PR TITLE
[s] Adds new Halloween content

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: New additions have been made for this Halloween and all future ones. Happy Halloween!
/:cl:

Silent tag is kept not to spoil it for the majority of players.

All new content  only loads during Halloween and is set to load during that time. Includes:
- Two Syndicate items. The hack-o'-lantern is a festive emag, and the candy corn crossbow is a festive ebow. Both maintain their original functionalities but have different aesthetic and can be ordered from the Holiday section of the uplink.
- Fixed all existing Halloween content from not working.

More content to come based on suggestions and my own ideas. Feedback appreciated!

### CODE REVIEW

File: code/modules/holiday/halloween.dm [hunk 1]
```diff
@@ -102,14 +102,10 @@
 
 	else if(trapped == SPOOKY_SKELETON)
 		visible_message("<span class='userdanger'><font size='5'>BOO!</font></span>")
-		playsound(loc, pick('sound/effects/xylophone1.ogg','sound/effects/xylophone2.ogg','sound/effects/xylophone3.ogg','sound/spookoween/girlscream.ogg'), 300, 1)
+		playsound(loc, 'sound/spookoween/girlscream.ogg', 300, 1)
 		trapped = 0
 		spawn(90)
```
Comment (ChangelingRain) [inline code/modules/holiday/halloween.dm]: QDEL_IN(trapped_mob, 90)

File: code/modules/holiday/halloween.dm [hunk 2]
```diff
@@ -130,21 +126,15 @@
 		visible_message("<span class='userdanger'><font size='5'>THIS BEING RADIATES PURE EVIL! YOU BETTER RUN!!!</font></span>")
 		playsound(loc, 'sound/hallucinations/wail.ogg', 300, 1)
 		var/mob/living/simple_animal/hostile/faithless/F = new(loc)
-		F.stance = HOSTILE_STANCE_ATTACK
-		F.GiveTarget(usr)
 		trapped = 0
 		spawn(120)
```
Comment (ChangelingRain) [inline code/modules/holiday/halloween.dm]: QDEL_IN(F, 120)

**Comment by bgobandit** [comment]
could my clown stuff possibly be rolled into this instead of 2 separate prs

**Comment by optimumtact** [comment]
@bgobandit ? that seems pointless

**Comment by Xhuis** [comment]
Yeah, I don't see a reason why that would be necessary.

File: code/modules/holiday/halloween.dm [hunk 3]
```diff
@@ -130,21 +126,15 @@
...
-				smoke.start()
+			if(F)
 				qdel(F)
 
 	else if(trapped == INSANE_CLOWN)
 		visible_message("<span class='userdanger'><font size='5'>...</font></span>")
 		playsound(loc, 'sound/spookoween/scary_clown_appear.ogg', 300, 1)
-		var/mob/living/simple_animal/hostile/retaliate/clown/insane/IC = new (loc)
-		IC.GiveTarget(usr)
+		new/mob/living/simple_animal/hostile/retaliate/clown/insane(loc)
```
Comment (optimumtact) [inline code/modules/holiday/halloween.dm]: use the spawn_atom_to_turf helper! it will verify loc is a turf

**Comment by Cheridan** [comment]
>Removed the delay before greeting a traitor; it could be circumvented by checking notes instantly.

pretty sure that the delay was intentionally added so the YOU ARE A TRAITOR doesn't get caught up in the rest of the roundstart spam

**Comment by optimumtact** [comment]
Stop bundling random unrelated fixes into your pr's dangit.

**Comment by Xhuis** [comment]
@optimumtact I figured it wasn't bad, but I can further atomize it if you really want.
@Cheridan Based on my experience, the antagonist greetings always show last regardless. If it's really that bad I can throw something in.

**Comment by PKPenguin321** [comment]
No xhuis it's got the delay so that you don't miss your traitor notification in the roundstart spam and play the round without realizing you're a traitor

**Comment by Xhuis** [comment]
I know, that's what I was referencing. No other gamemodes use the delay because it never seems to be problematic.

File: code/game/gamemodes/traitor/traitor.dm [hunk 4]
```diff
@@ -59,10 +59,9 @@
 
 /datum/game_mode/traitor/post_setup()
 	for(var/datum/mind/traitor in traitors)
+		greet_traitor(traitor)
 		forge_traitor_objectives(traitor)
```
Comment (optimumtact) [inline code/game/gamemodes/traitor/traitor.dm]: you're greeting before they have objectives? Wouldn't this cause a bug?

**Comment by optimumtact** [comment]
See, I could have merged this ages ago if you hadn't bundled in these unrelated fixes

you delayed your own PR unnecessarily

**Comment by Xhuis** [comment]
:(

**Comment by Xhuis** [comment]
Changes made, and a giant commented-out block of code that never worked was removed. You can merge this now, or I can add more stuff if anyone wants.

**Comment by optimumtact** [comment]
You've removed a bunch of other peoples changes as far as I can tell

**Comment by optimumtact** [comment]
OH hang on, seems the diff was bugged or something. Looks fine now

**Pull request merged**